README.md

```diff
@@ -1,18 +1,1481 @@
+  return factor.provider === 'OKTA' && factor.factorType === 'question';
+});
+
+questionFactor.verify({
+  answer: 'mayonnaise'
+});
+```
+
+###### All other factors
+
+```javascript
+var factor = transaction.factors.find(function(factor) {
+  return factor.provider === 'YOUR_PROVIDER' && factor.factorType === 'yourFactorType';
+});
+
+factor.verify();
+```
+
+### MFA_CHALLENGE
+
+The user must verify the factor-specific challenge.
+
+```javascript
+{
+  status: 'MFA_CHALLENGE',
+  expiresAt: '2014-11-02T23:39:03.319Z',
+  factorResult: 'WAITING', // or CANCELLED, TIMEOUT, or ERROR
+  user: {
+    id: '00ugti3kwafWJBRIY0g3',
+    profile: {
+      login: 'isaac@example.org',
+      firstName: 'Isaac',
+      lastName: 'Brock',
+      locale: 'en_US',
+      timeZone: 'America/Los_Angeles'
+    },
+  },
+  factor: {
+    id: 'smsigwDlH85L9FyQK0g3',
+    factorType: 'sms',
+    provider: 'OKTA',
+    profile: {
+      phoneNumber: '+1 XXX-XXX-6688'
+    }
+  },
+  verify: function(options) { /* returns another transaction */ },
+  poll: function() { /* returns another transaction */ },
+  prev: function() { /* returns another transaction */ },
+  cancel: function() { /* terminates the auth flow */ },
+  data: { /* the parsed json response */ }
+}
+```
+
+#### verify(options)
+
+  - `passCode` - OTP sent to device
+
+```javascript
+transaction.verify({
+  passCode: '615243'
+});
+```
+
+#### poll()
+
+Poll until factorResult is not WAITING. Throws AuthPollStopError if prev or cancel is called.
+
+```javascript
+transaction.poll();
+```
+
+#### prev()
+
+End current factor verification and return to `MFA_REQUIRED`.
+
+```javascript
+transaction.prev();
+```
+
+### SUCCESS
+
+The end of the authentication flow! This transaction contains a sessionToken you can exchange for an Okta cookie.
+
+```javascript
+{
+  expiresAt: '2015-06-08T23:34:34.000Z',
+  status: 'SUCCESS',
+  sessionToken: '00p8RhRDCh_8NxIin-wtF5M6ofFtRhfKWGBAbd2WmE',
+  user: {
+    id: '00uhm5QzwyZZxjrfp0g3',
+    profile: {
+      login: 'exampleUser@example.com',
+      firstName: 'Test',
+      lastName: 'User',
+      locale: 'en_US',
+      timeZone: 'America/Los_Angeles'
+    }
+  }
+}
+```
+
+## session.setCookieAndRedirect(sessionToken, redirectUri)
+
+This allows you to create a session using a sessionToken.
+
+  - `sessionToken` - Ephemeral one-time token used to bootstrap an Okta session.
+  - `redirectUri` - After setting a cookie, Okta redirects to the specified URI. The default is the current URI.
+
+```javascript
+authClient.session.setCookieAndRedirect(transaction.sessionToken);
+```
+
+## session.exists()
+
+Returns a promise that resolves with `true` or `false`.
+
+```javascript
+authClient.session.exists()
+.then(function(exists) {
+  if (exists) {
+    // logged in
+  } else {
+    // not logged in
+  }
+});
+```
+
+## session.get()
+
+Gets the active [session](http://developer.okta.com/docs/api/resources/sessions.html#example).
```

Comment:
I like the idea of linking to the AuthN API methods & docs - can we do this with everything else in this README?